test/jdk/com/sun/management/ThreadMXBean/ThreadAllocatedMemory.java

```diff
@@ -201,38 +206,102 @@ private static void testGetThreadsAllocatedBytes()
             } catch (InterruptedException e) {
-                System.out.println("Unexpected exception is thrown.");
+                System.out.println("Unexpected exception thrown.");
+                e.printStackTrace(System.out);
+                break;
+            }
+        }
+    }
+
+    private static void testGetAllThreadAllocatedBytes()
+        throws Exception {
+
+        // baseline should be positive
+        Thread curThread = Thread.currentThread();
+        long cumulative_size = mbean.getAllThreadAllocatedBytes();
+        if (cumulative_size <= 0) {
+            throw new RuntimeException(
+                "Invalid allocated bytes returned for " + curThread.getName() + " = " + cumulative_size);
+        }
+
+        // start threads
+        done = false; done1 = false;
+        for (int i = 0; i < NUM_THREADS; i++) {
+            threads[i] = new MyThread("MyThread-" + i);
+            threads[i].start();
+        }
+
+        // wait for threads to block after doing some allocation
+        waitUntilThreadsBlocked();
+
+        // check after threads are blocked
+        cumulative_size = checkResult(curThread, cumulative_size, mbean.getAllThreadAllocatedBytes());
+
+        // let threads go to do some more allocation
+        synchronized (obj) {
+            done = true;
+            obj.notifyAll();
+        }
+
+        // wait for threads to get going again. we don't care if we
+        // catch them in mid-execution or if some of them haven't
+        // restarted after we're done sleeping.
+        goSleep(400);
+
+        System.out.println("Done sleeping");
+
+        // check while threads are running
+        cumulative_size = checkResult(curThread, cumulative_size, mbean.getAllThreadAllocatedBytes());
+
+        // let threads exit
+        synchronized (obj) {
+            done1 = true;
+            obj.notifyAll();
+        }
+
+        for (int i = 0; i < NUM_THREADS; i++) {
+            try {
+                threads[i].join();
+            } catch (InterruptedException e) {
+                System.out.println("Unexpected exception thrown.");
                 e.printStackTrace(System.out);
                 break;
             }
         }
+
+        // check after threads exit
+        checkResult(curThread, cumulative_size, mbean.getAllThreadAllocatedBytes());
     }
 
-    private static void ensureValidSize(long size) {
+    private static void ensureValidSize(Thread curThread, long size) {
         // implementation could have started measurement when
         // measurement was enabled, in which case size can be 0
         if (size < 0) {
             throw new RuntimeException(
-                "Invalid allocated bytes returned = " + size);
+                "Invalid allocated bytes returned for thread " +
+                curThread.getName() + " = " + size);
         }
     }
 
-    private static void checkResult(Thread curThread,
+    private static long checkResult(Thread curThread,
```

Comment:
There is another `checkResult` below? Should they be replaced by a single method?

Reply:
No, there's only one checkResult method. I changed the result type from void to long.